src/main/java/com/ft/methodearticlemapper/transformation/MethodeLinksBodyProcessor.java

```diff
@@ -250,6 +256,116 @@ private void processATags(Map<Node, String> aTags, List<Content> content) {
         return content.stream().filter(c -> c.getUuid().equals(uuid)).findFirst();
     }
 
+    private void fixFormattingOfLinkText(Node node) {
+        Node pNode = node.getParentNode();
+        if (pNode != null) {
+            Node nodeBeforeATag = getNodeBeforeATag(pNode);
+            
+            // Adding space if needed, right before the <a> tag
+            if (nodeBeforeATag != null) {
+                String beforeATagText = nodeBeforeATag.getTextContent();              
+                if (!Strings.isNullOrEmpty(beforeATagText)) {
+                    String newTextBefore = fixImproperWhitespaceBeforeATag(beforeATagText);
+                    nodeBeforeATag.setTextContent(newTextBefore);
+                } 
+            }
+            
+            // Extracting punctuation outside the <a> tag
+            String linkText = node.getTextContent().trim();
+            String punctuation = getPunctuationFromLinkText(linkText);
+            Node nodeAfterATag = getNodeAfterATag(pNode);
+
+            if (punctuation != null) {
+                String newLinkText = linkText.replaceAll("[" + punctuation + "]+$", "");
```

Comment:
Would it be simpler to just strip off `punctuation.length()` characters from the end of `linkText`? The way it's obtained, it is really a literal string that definitely exists as the final substring of `linkText`, rather than a character class, so the square brackets are just confusing (though probably necessary if you do it this way).